SnakeGame/Queue.js

```diff
@@ -0,0 +1,36 @@
+var Queue = function () {
+    this.first = null;
+    this.size = 0;
+};
+
+var Node = function (data) {
+    this.data = data;
+    this.next = null;
+};
+
+Queue.prototype.enqueue = function (data) {
+    var node = new Node(data);
+
+    if (!this.first) {
+        this.first = node;
+    } else {
+        let n = this.first;
+        while (n.next) {
```

Comment:
This loop makes this Queue data structure inefficient -- but this is okay for such a small data structure. In effect, this is no different (in terms of time complexity) as implementing the queue using `Array::push()` and `Array::shift()`